### DISCUSSION TITLE
Warning when running `train_segmentation_model` (Hugging Face unauthenticated requests)

### DISCUSSION THREAD

**Discussion by ihahachi** [opengeos/geoai#695] [Q&A]
hi everyone,
When running the `train_segmentation_model` function, the training process starts correctly, but I consistently receive the following warning:

```
Warning: You are sending unauthenticated requests to the HF Hub. Please set a **HF_TOKEN** to enable higher rate limits and faster downloads. 
WARNING:huggingface_hub.utils._http:Warning: You are sending unauthenticated requests to the HF Hub. Please set a **HF_TOKEN** to enable higher rate limits and faster downloads.
```

### Code Used

```python
geoai.train_segmentation_model(
    images_dir=f"{out_folder}/images",
    labels_dir=f"{out_folder}/labels",
    output_dir=f"{out_folder}/unet_models",
    architecture="unet",
    encoder_name="resnet34",
    encoder_weights="imagenet",
    num_channels=3,
    num_classes=2,
    batch_size=8,
    num_epochs=20,
    learning_rate=0.001,
    val_split=0.2,
)
```

### Observed Behavior

* The model training runs successfully.
* However, the warning appears multiple times during execution.
* It seems related to downloading pretrained weights (`encoder_weights="imagenet"`) via Hugging Face Hub.

### Expected Behavior

* Either:

  * No warning should appear for standard usage, or
  * Clear documentation should indicate that authentication is required when using pretrained encoders.

### Questions

1. Is this warning expected when using pretrained encoders like `"resnet34"` with `"imagenet"` weights?
2. Does this affect performance, caching, or download reliability?
3. Is setting a `HF_TOKEN` required or just recommended?
4. Should the library handle this internally or suppress the warning?

### Environment

* Google Colab
* Python 3

**Comment by khalilurrrahmanridoykhan** [comment]
Yes, this warning is expected and is not a training failure. `encoder_weights="imagenet"` requests pretrained public weights; Hugging Face Hub permits unauthenticated public downloads but applies lower rate limits. Once downloaded, the file is normally reused from the local Hub cache, so the warning does not change the model's mathematics or training accuracy.

Authentication is **recommended**, not required, for a public model. In Colab, store a read token in the Colab Secrets panel (for example under `HF_TOKEN`) and authenticate before creating the model:

```python
from google.colab import userdata
from huggingface_hub import login

login(token=userdata.get("HF_TOKEN"), add_to_git_credential=False)
```

Alternatively, set the `HF_TOKEN` environment variable before calling `train_segmentation_model`. Do not paste the token directly into a shared notebook or commit it to Git. The official Hub authentication guide documents both login and token-based authentication: https://huggingface.co/docs/huggingface_hub/package_reference/authentication

If pretrained weights are not wanted, use:

```python
encoder_weights=None
```

That avoids the pretrained-weight download, but it also trains the encoder from random initialization and may require more data/training. Suppressing the warning inside GeoAI would hide useful information when rate limiting or a fresh runtime cache causes download problems, so documentation is preferable to library-level suppression.
